src/app/client/src/app/modules/core/services/device-register/device-register.service.ts

```diff
@@ -1,19 +1,25 @@
-import { Injectable } from '@angular/core';
+import { Injectable, OnDestroy } from '@angular/core';
 import { PublicDataService } from './../public-data/public-data.service';
-import { ConfigService, RequestParam,  HttpOptions} from '@sunbird/shared';
+import { ConfigService,  HttpOptions} from '@sunbird/shared';
 import * as moment from 'moment';
 import { UUID } from 'angular2-uuid';
 import { HttpClient } from '@angular/common/http';
 import { DeviceDetectorService } from 'ngx-device-detector';
+import { Observable, timer, Subscription } from 'rxjs';
 
 @Injectable({
   providedIn: 'root'
 })
-export class DeviceRegisterService {
+export class DeviceRegisterService implements OnDestroy {
```

Comment:
why do we have OnDestroy event in service. Dose angular call life cycle hook on services too.

Reply:
OnDestroy removed , seems to be not needed, lift cycle hooks works in services